### PR TITLE
chore(NODE-6382): add benchmarking for helpers

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

- Adds a benchmark suite that will output to our trendcharting tool.

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

- There are a number of code paths that don't relate to serialize/deserialize but can still be useful for critical paths in an application. It would be useful to improve those where we can and ensure we maintain the improvements where they are made.
- Help with the change proposed: https://github.com/mongodb/js-bson/pull/708

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/bench/custom/main.mjs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  .on('cycle', function logBenchmark(event) {
+    console.log(String(event.target));
+  })
+  .on('complete', function outputPerfSend() {
+    const data = Array.from(this).map(bench => ({
+      info: { test_name: bench.name },
+      metrics: [{ name: 'ops_per_sec', value: bench.hz }]
+    }));
+    console.log(util.inspect(data, { depth: Infinity, colors: true }));
+    fs.writeFileSync('results.json', JSON.stringify(data), 'utf8');
```
Comment (W-A-James) [inline test/bench/custom/main.mjs:37]: Can we rename this to something like `customBenchmarkResults.json` so it's easier to check the results and build tables locally if need be?

**Review by W-A-James — CHANGES_REQUESTED** [review]

File: test/bench/custom/main.mjs (hunk 1 above)
Comment (nbbeeken) [reply test/bench/custom/main.mjs]: done!

**Review by W-A-James — DISMISSED** [review]

File: test/bench/custom/benchmarks.mjs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable strict */
+import { BSON } from '../../../lib/bson.mjs';
+
+const ObjectId_isValid = [
+  function objectid_isvalid_strlen() {
+    BSON.ObjectId.isValid('a');
+  },
+  function objectid_isvalid_bestcase_false() {
```
Comment (dariakp) [inline test/bench/custom/benchmarks.mjs:8]: what makes this the best case? and the other the worst case?

Comment (nbbeeken) [reply test/bench/custom/benchmarks.mjs]: Most or least short-circuiting. Updated the test names and added comments to clarify!

Comment (dariakp) [reply test/bench/custom/benchmarks.mjs]: Test case names lgtm now - thanks!

**Review by W-A-James — APPROVED** [review]

**Pull request merged**